x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/registry/ModelRegistry.java

```diff
@@ -657,7 +681,7 @@ public void deleteModel(String inferenceEntityId, ActionListener<Boolean> listen
 
         DeleteByQueryRequest request = new DeleteByQueryRequest().setAbortOnVersionConflict(false);
         request.indices(InferenceIndex.INDEX_PATTERN, InferenceSecretsIndex.INDEX_PATTERN);
-        request.setQuery(documentIdQuery(inferenceEntityId));
+        request.setQuery(documentIdsQuery(inferenceEntityIds));
```

Comment:
Interesting side not, delete by query will not fail if it can't find an id. Which is helpful here because for the revoking of authorization situation, we don't really care if the model exists already or not, just that it is removed if it did exist.